pkg/cloudprovider/provider/vsphere/provider.go

```diff
@@ -590,74 +607,51 @@ func (p *provider) Get(machine *v1alpha1.Machine, _ *cloudprovidertypes.Provider
 	if err != nil {
 		return nil, fmt.Errorf("failed to get datacenter finder: %v", err)
 	}
-	virtualMachine, err := finder.VirtualMachine(ctx, machine.Spec.Name)
+	virtualMachineList, err := finder.VirtualMachineList(ctx, machine.Spec.Name)
```

Comment:
We can not re-use `p.Get` here because it just returns a single machine and doesn't care about the creation complete part, but we need all machines here and we need to destroy the ones whose creation didn't complete